app/src/main/java/com/x8bit/bitwarden/data/autofill/fido2/model/Fido2RegisterCredentialResult.kt

```diff
@@ -0,0 +1,24 @@
+package com.x8bit.bitwarden.data.autofill.fido2.model
+
+/**
+ * Models the data returned from creating a FIDO 2 credential.
+ */
+sealed class Fido2RegisterCredentialResult {
+
+    /**
+     * Indicates the credential has been successfully registered.
+     */
+    data class Success(
+        val registrationResponse: String,
+    ) : Fido2RegisterCredentialResult()
+
+    /**
+     * Indicates there was an error and the credential was not registered.
+     */
+    data object Error : Fido2RegisterCredentialResult()
+
+    /**
+     * Indicates the user cancelled the request.
+     */
+    data object Cancelled : Fido2RegisterCredentialResult()
```

Comment:
Was Cancelled handled via a specific exception previously?

Reply:
Yes, cancelled was previously handled by constructing and passing a `CreateCredentialCancellationException` along with the `Error` result.